crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2915,7 +2915,7 @@ pub async fn list_payment_methods(
         .transpose()?;
     let setup_future_usage = payment_intent.as_ref().and_then(|pi| pi.setup_future_usage);
     let payment_type = payment_attempt.as_ref().map(|pa| {
-        let amount = api::Amount::from(pa.amount);
+        let amount = api::Amount::from(pa.net_amount.get_order_amount());
```

Comment:
Only order amount needs to be used here, we can refer intent amount itself.

Reply:
yes, but here it's inside payment_attempt's map block. payment_intent is also option here